packages/thrift-server-core/src/main/parse_json.ts

```diff
@@ -0,0 +1,257 @@
+        // or a word.
+
+        this.white()
+        switch (this.getCh()) {
+            case '{':
+                return this.object()
+            case '[':
+                return this.array()
+            case '"':
+                return this.string()
+            case '-':
+                return this.number()
+            default:
+                return this.getCh() >= '0' && this.getCh() <= '9'
+                    ? this.number()
+                    : this.word()
+        }
+    }
+
+    private next(c?: string) {
+        // If a c parameter is provided, verify that it matches the current character.
+        if (c && c !== this.getCh()) {
+            throw new SyntaxError("Expected '" + c + "' instead of '" + this.getCh() + "'")
+        }
+
+        // Get the next character. When there are no more characters,
+        // return the empty string.
+
+        this.ch = this.text.charAt(this.at)
+        this.at += 1
+        return this.ch
+    }
+
+    private getCh() {
+        return this.ch
+    }
+
+    private number() {
+        // Parse a number value.
+        let number = 0
+        let string = ''
+
+        if (this.getCh() === '-') {
+            string = '-'
+            this.next('-')
+        }
+        while (this.getCh() >= '0' && this.getCh() <= '9') {
+            string += this.getCh()
+            this.next()
+        }
+        if (this.getCh() === '.') {
+            string += '.'
+            while (this.next() && this.getCh() >= '0' && this.getCh() <= '9') {
+                string += this.getCh()
+            }
+        }
+        if (this.getCh() === 'e' || this.getCh() === 'E') {
+            string += this.getCh()
+            this.next()
+            if (this.getCh() === '-' || this.getCh() === '+') {
+                string += this.getCh()
+                this.next()
+            }
+            while (this.getCh() >= '0' && this.getCh() <= '9') {
+                string += this.getCh()
+                this.next()
+            }
+        }
+        number = +string
+        if (!isFinite(number)) {
+            throw new SyntaxError('Bad number')
+        } else if (number >= Int64.MAX_INT || number <= Int64.MIN_INT) {
+            // Return raw string for further process in JSONProtocol
+            return string
+        } else {
+            return number
+        }
+    }
+
+    private string(): string {
+        // Parse a string value.
+        let hex = 0
+        let i = 0
+        let string = ''
+        let uffff = 0
+
+        // When parsing for string values, we must look for " and \ characters.
+        if (this.getCh() === '"') {
+            while (this.next()) {
+                if (this.getCh() === '"') {
+                    this.next()
+                    return string
+                }
+                if (this.getCh() === '\\') {
+                    this.next()
+                    if (this.getCh() === 'u') {
+                        uffff = 0
```

Comment:
could consider supporting values like '\u{0061}' which I guess is supported in ES6